### DISCUSSION TITLE
ReFlashing SNES repro 29LV640 - UNKNOWN FLASHROM

### DISCUSSION THREAD

**Discussion by StanMOI** [sanni/cartreader#141] [Need help]
Hi Sanni,

I'am trying to Reflashing SNES Repro with MX29LV640ETTI flash chip.
I get this error : 
![image](https://user-images.githubusercontent.com/38383423/116742028-19cfd800-a9f7-11eb-89a5-a8bc0fddc459.png)

Here is my configuration:
![image](https://user-images.githubusercontent.com/38383423/116741725-c78eb700-a9f6-11eb-849d-f320936e84f7.png)
Switched to 3v (vs 5v)
I tried with 3v and 5v  = no effect , same error message.

![IMG_5910](https://user-images.githubusercontent.com/38383423/116790934-4ba87380-aab7-11eb-9c8d-7355fd871029.jpg)
![IMG_5909](https://user-images.githubusercontent.com/38383423/116790678-ed2ec580-aab5-11eb-9815-f1fa7a9c1663.jpg)
![IMG_5911](https://user-images.githubusercontent.com/38383423/116790772-5d3d4b80-aab6-11eb-9c74-df1962df357a.jpeg)
![IMG_5912](https://user-images.githubusercontent.com/38383423/116790774-69c1a400-aab6-11eb-84f1-a55cce736745.jpeg)

Here is my SNES PCB Repro ROM information that I want to reflash:

![IMG_5913](https://user-images.githubusercontent.com/38383423/116796930-cdf95d80-aae0-11eb-9e03-daba97db75eb.jpg)

When I dump game, with command "READ ROM" = its works fine.
When I do "READ SAVE" = get message "Does not have SRAM" (maybe its because I never played and dont have any saves on SRAM)
When I do "Write SAVE" = get message "Does not have SRAM"

Cartridge fingers was cleared with IPA, make no changes.

I re-used the 5V regulator desoldered in "step 1" 
<img width="799" alt="Capture d’écran 2021-05-02 à 01 01 00" src="https://user-images.githubusercontent.com/38383423/116797134-92f82980-aae2-11eb-8f14-d0dc2b4d396d.png">

I see that you have code for those chip, but its not working :(

> }
>   else if ((strcmp(flashid, "C2C9") == 0) || (strcmp(flashid, "C2CB") == 0)) {
>     println_Msg(F("MX29LV640 detected"));
>     println_Msg(F("ATTENTION 3.3V"));
>     flashSize = 8388608;
>     flashromType = 2;
>   }

I have two boards like this one (LoROM and HiRom), same problem for two boards.

Can you help please.

**Comment by sanni** [comment]
Remove this solder bridge and try again:

![image](https://i.imgur.com/ZAEYL8f.jpg)

**Reply by StanMOI** [reply]
> Remove this solder bridge and try again:
> 
> ![image](https://camo.githubusercontent.com/1866edd477062327d52f08dec086af5a662ade1791e89be54d3740345ba03704/68747470733a2f2f692e696d6775722e636f6d2f5a4145594c38662e6a7067)

YEAH ! That's it !
It's works now!
Thank you.

For what for this solder bridge? 
Do you know what pinouts connections for this jumper ?
Do i have to re-solder it back?

**Comment by sanni** [comment]
It's a write protection. It connects the WE pin of the flashrom to VCC and therefore disabling any writes to the flashrom. You don't have to solder it back.

**Reply by StanMOI** [reply]
> It's a write protection. It connects the WE pin of the flashrom to VCC and therefore disabling any writes to the flashrom. You don't have to solder it back.


Ok,
Can I always leave WE pin of flashrom connected To to /WR (pin 54) of cart edge?

**Comment by sanni** [comment]
You can't have both the cable and the bridge at the same time. So you can either leave the cable soldered and remove the solder bridge or remove the cable and solder the bridge again.

I would do the first option.

**Reply by StanMOI** [reply]
> You can't have both the cable and the bridge at the same time. So you can either leave the cable soldered and remove the solder bridge or remove the cable and solder the bridge again.
> 
> I would do the first option.

👌🏻
Just for information,
« /WR » pin 54 of cart age in GND signal ? Normally LOW signal, yes?
